### PR TITLE
Improve device locale + user override separation

### PR DESCRIPTION
**Pull request by tstirrat**
Separates the device detection + initial set, locale setting code, and user override flags in a way that makes intent very clear.

### CODE REVIEW

**Comment by tstirrat** [comment]
:lock: PR author to merge via `git-multi-pr merge`

File: app/locales/languages.js [hunk 1]
```diff
@@ -31,44 +30,50 @@ import zh_Hant from './zh-Hant.json';
...
+  return normalizeLocale(fullLocale);
+}
 
-      // Run state updating callback to trigger rerender
-      typeof callback === 'function' ? callback(userLang) : null;
+/**
+ * @param {string} fullLocale
+ */
+function normalizeLocale(fullLocale) {
+  const [first] = fullLocale.split('-');
```
Comment (kenpugsley) [inline app/locales/languages.js:59]: Is this a typo?  Shouldn't it be `fullLocale.split('_');`

Comment (tstirrat) [reply app/locales/languages.js:59]: Good catch, I better double check this tomorrow

Comment (tstirrat) [reply app/locales/languages.js]: I was supposed to be normalizing `en_US` to `en-us` for dayjs. Fixed.

**Comment by tstirrat** [comment]
:unlock: Merging via `git-multi-pr merge`

**Comment by tstirrat** [comment]
:dash:

**Pull request merged**